Source/WebCore/Modules/webaudio/AudioContext.cpp

```diff
@@ -187,6 +189,20 @@ double AudioContext::baseLatency()
     return static_cast<double>(destination().framesPerBuffer()) / sampleRate();
 }
 
+double AudioContext::outputLatency()
+{
+    lazyInitialize();
+
+    if (noiseInjectionPolicies())
+        return 512 / sampleRate(); // A fixed, but reasonable value for most platforms.
+
+    return (MediaTime { static_cast<int64_t>(destination().outputLatency()), static_cast<uint32_t>(sampleRate()) }
+#if USE(AUDIO_SESSION)
+            + MediaTime { static_cast<int64_t>(AudioSession::protectedSharedSession()->outputLatency()), static_cast<uint32_t>(AudioSession::protectedSharedSession()->sampleRate()) }
```

Comment:
Would it be better to leave destination do this computation?
That way, we can do the computation in GPUProcess and remove the need to send to WebProcess the audio session latency.

Reply:
Doing so would require to have a way for the RemoteAudioDestination/RemoteAudioDestinationProxy be notified when the AudioSession changes. It's a much bigger change.

And currently the RemoteAudioSession in the WebContent is being notified of changes of AudioSession in the GPU process. So I re-used that.

it made the change much smaller.

I could move that arithmetic in the AudioDestination, but then it kind of blur which latency are we talking about. The AudioDestination itself, the AudioSession, both?

Reply:
I would prefer we would move the computation to the AudioDestination.
It is probably the audio destination that will handle setSinkId computation as well, so that seems better this way.